components/nimbus/src/stateful/behavior.rs

```diff
@@ -427,6 +429,13 @@ impl EventQueryType {
         })
     }
 
+    pub fn validate_query(maybe_query: &str) -> Result<bool> {
+        let regex = Regex::new(
+            r#"^["'][^"']+["']\|event(?:Sum|LastSeen|CountNonZero|Average|AveragePerNonZeroInterval)\(["'](?:Years|Months|Weeks|Days|Hours|Minutes)["'], \d+(?:, \d+)?\)$"#,
+        )?;
```

Comment:
> Note that regex compilation tends to be a somewhat expensive process, and unlike higher level environments, compilation is not automatically cached for you. One should endeavor to compile a regex once and then reuse it. For example, it’s a bad idea to compile the same regex repeatedly in a loop.

If we want to cache this we should use https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html or https://doc.rust-lang.org/stable/std/sync/struct.OnceLock.html (threadsafe).

OR if we decide to only enable this check in debug builds, then we can take the hit, it doesn't matter